blaze-client/src/main/scala/org/http4s/blaze/client/PoolManager.scala

```diff
@@ -109,13 +110,13 @@ private final class PoolManager[F[_], A <: Connection[F]](
 
   private def incrConnection(key: RequestKey): F[Unit] =
     F.delay {
-      curTotal += 1
+      curTotal.getAndIncrement()
       allocated.update(key, allocated.getOrElse(key, 0) + 1)
```

Comment:
I would suppose that `allocated.update(key, allocated.getOrElse(key, 0) + 1)` is also not thread safe, is it?
Because it is just
```
private val allocated = mutable.Map.empty[RequestKey, Int]
```
Shouldn't `scala.collection.concurrent.TrieMap` be used here instead?

Reply:
Probably you're right, but I feel that changing the collection for sure will need some performance checking. I'd rather leave this PR with only the proposed change. Feel free to open a new one, or I may work on it after a while.

Reply:
Yes, that's certainly incorrect.

Reply:
The semaphore should protect us from concurrent updates, but I think cached reads are still a problem.